### PR TITLE
feature: add custom network dialog

### LINKED ISSUE #2911: [Demand] Add customized network (RPC)
## Catalogue

- [ ] UI/UX
- [x] Mask Core Feature
- [ ] Mask Plugin Feature
- [ ] Security/Cryptography Feature
- [ ] Other
  - /_List it here_/

## Part 1: Introduction and Background

A screenshot from MetaMask:

image

For Mask Network, we can replace the coming soon into add custom RPC.

image

## Part 2: Problem to Solve

+ Remove hardcoded Ether as the default currency symbol. The currency symbol should be chain-agnostic and modifiable.
+ Plugins that need to interact with the chain should declare their supported chain ids. If an unknown chain id is picked, it should present a warning message to the user and refuse to render UI.

## Notes

This feature may be only available in the dev mode in the early days.

### PR DESCRIPTION
**Pull request by relaxgo**
closes #2911

### CODE REVIEW

File: packages/maskbook/src/extension/background-script/EthereumServices/provider.ts [hunk 1]
```diff
@@ -49,6 +49,9 @@ export async function createWeb3() {
             return await MetaMask.createWeb3()
         case ProviderType.WalletConnect:
             return WalletConnect.createWeb3()
+        case ProviderType.CustomNetwork:
+            // TODO  connect customNetwork
```
Comment (septs) [inline packages/maskbook/src/extension/background-script/EthereumServices/provider.ts:53]: add an issue number?

File: packages/maskbook/src/extension/background-script/EthereumServices/sign.ts [hunk 2]
```diff
@@ -21,6 +21,9 @@ export async function sign(data: string, address: string) {
         case ProviderType.MetaMask:
         case ProviderType.WalletConnect:
             return web3.eth.personal.sign(data, address, '')
+        case ProviderType.CustomNetwork:
+            // TODO handle customNetwork
```
Comment (septs) [inline packages/maskbook/src/extension/background-script/EthereumServices/sign.ts:25]: add an issue number?

File: packages/maskbook/src/extension/background-script/EthereumServices/signTransaction.ts [hunk 3]
```diff
@@ -29,6 +29,9 @@ export async function signTransaction(config: TransactionConfig) {
                     else resolve(signed.raw)
                 })
                 break
+            case ProviderType.CustomNetwork:
+                //TODO handle custom network
```
Comment (septs) [inline packages/maskbook/src/extension/background-script/EthereumServices/signTransaction.ts:33]: add an issue number?

**Comment by guanbinrui** [comment]
replaced by #2985

**Pull request closed without merging**